File: app/src/main/java/com/wa2c/android/medoly/plugin/action/lrclyrics/activity/MainActivity.java
package com.wa2c.android.medoly.plugin.action.lrclyrics.activity;

import android.app.Activity;
import android.content.Intent;
import android.widget.Button;

import com.wa2c.android.medoly.library.MedolyEnvironment;
import com.wa2c.android.medoly.plugin.action.lrclyrics.R;
import com.wa2c.android.medoly.plugin.action.lrclyrics.util.AppPrefs_;
import com.wa2c.android.medoly.plugin.action.lrclyrics.util.AppUtils;

import org.androidannotations.annotations.AfterViews;
import org.androidannotations.annotations.Click;
import org.androidannotations.annotations.EActivity;
import org.androidannotations.annotations.ViewById;
import org.androidannotations.annotations.sharedpreferences.Pref;

/**
 * Main activity.
 */
@EActivity(R.layout.activity_main)
public class MainActivity extends Activity {

    @Pref
    AppPrefs_ appPrefs;

    @ViewById
    Button launchMedolyButton;

    @Click(R.id.launchSearchButton)
    void launchSearchButtonClick() {
        startActivity(new Intent(MainActivity.this, SearchActivity_.class));
    }

    @Click(R.id.launchCacheButton)
    void setLaunchCacheButtonClick() {
        startActivity(new Intent(MainActivity.this, CacheActivity_.class));
    }

    @Click(R.id.launchSettingsButton)
    void setLaunchSettingsButtonClick() {
        startActivity(new Intent(MainActivity.this, SettingsActivity.class));
    }

    @Click(R.id.launchMedolyButton)
    void setLaunchMedolyButtonClick() {
        Intent intent = getPackageManager().getLaunchIntentForPackage(MedolyEnvironment.MEDOLY_PACKAGE);
        if (intent == null) {
            AppUtils.showToast(this, R.string.message_no_medoly);
            return;
        }
        startActivity(intent);
    }

    @AfterViews
    void afterViews() {
        AppUtils.versionUp(this);

        android.app.ActionBar actionBar = getActionBar();
        if (actionBar != null) {
            actionBar.setDisplayShowHomeEnabled(true);
            actionBar.setDisplayShowTitleEnabled(true);
        }
    }

}

File: app/src/main/java/com/wa2c/android/medoly/plugin/action/lrclyrics/activity/MainActivity.kt
package com.wa2c.android.medoly.plugin.action.lrclyrics.activity

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.wa2c.android.medoly.library.MedolyEnvironment
import com.wa2c.android.medoly.plugin.action.lrclyrics.R
import com.wa2c.android.medoly.plugin.action.lrclyrics.databinding.ActivityMainBinding
import com.wa2c.android.medoly.plugin.action.lrclyrics.util.AppUtils


/**
 * Main activity.
 */
class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)

        supportActionBar?.let {
            it.setDisplayShowHomeEnabled(true)
            it.setDisplayShowTitleEnabled(true)
            it.setIcon(R.drawable.ic_launcher)
        }

        // Search
        binding.launchSearchButton.setOnClickListener {
            startActivity(Intent(this, SearchActivity::class.java))
        }

        // Cache
        binding.launchCacheButton.setOnClickListener {
            startActivity(Intent(this, CacheActivity::class.java))
        }

        // Settings
        binding.launchSettingsButton.setOnClickListener {
            startActivity(Intent(this, SettingsActivity::class.java))
        }

        // Launch Medoly
        binding.launchMedolyButton.setOnClickListener {
            val intent = packageManager.getLaunchIntentForPackage(MedolyEnvironment.MEDOLY_PACKAGE)
            if (intent == null) {
                AppUtils.showToast(this, R.string.message_no_medoly)
                return@setOnClickListener
            }
            startActivity(intent)
        }
    }

}
